### ISSUE TITLE
Virtual memory limits too low by default, add documentation for vm.max_map_count

### ISSUE THREAD

**Issue by davidjmeier** [deviantony/docker-elk#354]
### Problem description

On most linux systems the `vm.max_map_count` will be too low resulting in the following error when starting Elasticsearch: `[1]: max virtual memory areas vm.max_map_count [65536] is too low, increase to at least [262144]`

The README should reflect the `sysctl` command and reference the document on elastic.co.

### Extra information

The command to increase `vm.max_map_count` immediately on a running system is:
`[sudo] sysctl -w vm.max_map_count=262144`

Setting the value using systctl will not persist a reboot.  To have this value set at boot modify `/etc/sysctl.conf` to include `vm.max_map_count=262144`.  Users can validate this is the setting applied by using the `sysctl vm.max_map_count` command.

This is outlined on the Elastic site at: [Elastic.co - Virtual Memory](https://www.elastic.co/guide/en/elasticsearch/reference/current/vm-max-map-count.html)

#### Stack configuration

N/A

#### Docker setup

N/A

#### Docker logs
If docker-compose is run interactively the elasticsearch container will start then quit.  One of the side effects of this happening is when trying to access Kibana it will state: `Kibana server is not ready yet`.

`elasticsearch_1  | [1]: max virtual memory areas vm.max_map_count [65536] is too low, increase to at least [262144]`

**Comment by antoineco** [comment]
It used to be documented, but we switched to a single node setup for simplicity (see 586302bf4e43a474ab7dbbbcafbd033f2b150589).

The corresponding docs moved to the wiki, and is referenced inside the README: https://github.com/deviantony/docker-elk/wiki/Elasticsearch-cluster

Is there anything else we should do?

**Comment by davidjmeier** [comment]
Makes sense, I didn't see that in the ES cluster docu.  I only posted it because I would guess it comes up from time to time although it's an obvious fix.  I'm not sure what you mean by "we switched to a single node setup" as I don't believe that has any bearing on the issue cropping up for native Linux users.

On a standard Ubuntu 18.04 system:
```
user@mouth:~$ sysctl vm.max_map_count
vm.max_map_count = 65530
user@mouth:~$ uname -a
Linux mouth 4.15.0-43-generic #46-Ubuntu SMP Thu Dec 6 14:45:28 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
user@mouth:~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.1 LTS
Release:	18.04
Codename:	bionic
```

The container will fail/exit, by default.  I'm not sure if, in the past, the container would warn but run.

**Comment by antoineco** [comment]
What I meant is that tuning the `vm.max_map_count` setting is mandatory as per the ES docs, however this requirement is relaxed when you run in single-node mode.

Because the typical user of this project wants to evaluate the ELK stack before diving into a more production-ready setup, we decided to keep things as simple as possible and avoid confusing newcomers with optional OS tuning considerations. The next step in the journey is obvious to head to the Elastic docs.

**Comment by davidjmeier** [comment]
Thanks for clarifying.  Node being overly generic I thought you were implying single node operations with regard to Docker.  Regardless, this happens by default.  When using your repo, I just cloned and ended up in that state.  Not sure why what I'm doing would be any different.  I just saw it as a default blocker to people just using the repo as it stands.  Some might not know Docker well enough to find the error and just assume it's broken.

**Comment by antoineco** [comment]
Oh, if you saw it by default it's different. Maybe this warning was introduced recently enough that nobody here really noticed.

The real question is, considering the purpose I explained above (product evaluation, non-productive setups), should we take the warning seriously? Is it worth the extra step? What limitations are users going to run into if they don't consider setting this kernel parameter?

**Comment by davidjmeier** [comment]
I ended up getting a couple terminal session mixed up.  What I've figured out so far is the Ubuntu systems are unaffected and work successfully as I've tried on a few different versions of Docker with 18.04 fully patched and your compose and stack files work as expected.  It turns out the issue is on a Mint 19 machine.  I'll close this and dig into it to see what I can come up with, but I'm guessing in one of the builds something doesn't get set appropriately based on the alternate system name.
